File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/data/entity/impl/GameState.java
package com.dkellycollins.triggerfinger.data.entity.impl;

import com.dkellycollins.triggerfinger.data.entity.IGameState;

import java.io.Serializable;

public class GameState implements IGameState, Serializable {

    private boolean _gameOver;
    private boolean _isPaused;

    public GameState(boolean gameOver, boolean isPaused) {
        _gameOver = gameOver;
        _isPaused = isPaused;
    }

    @Override
    public boolean IsGameOver() {
        return _gameOver;
    }

    @Override
    public boolean IsPaused() {
        return _isPaused;
    }

    public void IsGameOver(boolean gameOver) {
        _gameOver = gameOver;
    }

    public void IsPaused(boolean isPaused) {
        _isPaused = isPaused;
    }
}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/data/entity/impl/GameState.kt
package com.dkellycollins.triggerfinger.data.entity.impl

import com.dkellycollins.triggerfinger.data.entity.IGameState

import java.io.Serializable

data class GameState(private var _gameOver: Boolean, private var _isPaused: Boolean) : IGameState, Serializable {

    override fun IsGameOver(): Boolean {
        return _gameOver
    }

    override fun IsPaused(): Boolean {
        return _isPaused
    }

    fun IsGameOver(gameOver: Boolean) {
        _gameOver = gameOver
    }

    fun IsPaused(isPaused: Boolean) {
        _isPaused = isPaused
    }
}
